openfasoc/generators/ldo-gen/simulations/templates/post_processing.py

```diff
@@ -0,0 +1,140 @@
+import ltspice
+import statistics
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+
+cap_list = ["1p", "5p"]
+cap_len = len(cap_list)
+fig1, axes1 = plt.subplots(len(cap_list), sharex=True, sharey=True)
+fig2, axes2 = plt.subplots(len(cap_list))
+fig3, axes3 = plt.subplots(len(cap_list))
+fig4, axes4 = plt.subplots(len(cap_list))
+fig5, axes5 = plt.subplots(len(cap_list))
+# fig6 , axes6 = plt.subplots(3)
+for i in range(cap_len):
+    l = ltspice.Ltspice(cap_list[i] + "_" + "cap_output.raw")
+    l.parse()
+    # get data
+    VREG = l.get_data("v(vreg)")
+    VREF = l.get_data("v(vref)")
+    cmp_out = l.get_data("v(cmp_out)")
+    clk = l.get_data("v(clk)")
+    # ctrl0    = l.get_data('v(ctrl_out[0])')
+    # ctrl1    = l.get_data('v(ctrl_out[1])')
+    # ctrl2    = l.get_data('v(ctrl_out[2])')
+    # ctrl3    = l.get_data('v(ctrl_out[3])')
+    # ctrl4    = l.get_data('v(ctrl_out[4])')
+    # ctrl5    = l.get_data('v(ctrl_out[5])')
+    # ctrl6    = l.get_data('v(ctrl_out[6])')
+    # ctrl7    = l.get_data('v(ctrl_out[7])')
+    # ctrl8    = l.get_data('v(ctrl_out[8])')
+    time = l.get_time()
+
+    VREG_list = VREG.tolist()  # converts numpy array to list
+    VREF_list = VREF.tolist()
+    time_list = time.tolist()
+
+    VREG_ripple_list = VREG_list[-9:]
+    VREF_ripple_list = VREF_list[-9:]
+    time_ripple_list = time_list[-9:]
+    VREG_min = min(VREG_ripple_list)
+    VREG_max = max(VREG_ripple_list)
+    VREG_ripple = VREG_max - VREG_min
+
+    axes1[i].set_title("VREG vs Time" + " " + str(cap_list[i]))
+    axes1[i].ticklabel_format(style="sci", axis="x", scilimits=(-6, -6))
+    axes1[i].plot(time_list, VREG_list, label="VREG")
+    axes1[i].plot(time_list, VREF_list, label="VREF")
+    axes1[i].legend(loc="lower right")
+    fig1.text(0.5, 0.04, "Time [us]", ha="center")
+    fig1.text(0.04, 0.5, "Vreg and Vref [V]", va="center", rotation="vertical")
+
+    # num_of_switches = ((ctrl0+2*ctrl1+4*ctrl2+8*ctrl3+16*ctrl4+32*ctrl5+64*ctrl6+128*ctrl7+256*ctrl8)/3.3)
```

Comment:
Do you need this?

Reply:
Was used during debugging to check the number of switches that are On at any given time.

Reply:
Maybe it should be turned on all the time?

Reply:
Yes it should be.